components/SerialServer/src/serial.c

```diff
@@ -176,7 +218,7 @@ static int is_newline(const uint8_t *c)
     return (c[0] == '\r' && c[1] == '\n') || (c[0] == '\n' && c[1] == '\r');
 }
 
-static int active_client = 0;
+static int active_client = -2;
```

Comment:
Maybe make `-1` and `-2` a dedicated define now, so this is a bit more transparent, what magic values exist.